spec/controllers/staff/organistations_controller_spec.rb

```diff
@@ -0,0 +1,13 @@
+require "rails_helper"
+
+RSpec.describe Staff::OrganisationsController do
+  context "when the user is not logged in" do
+    it "redirects the user to the root path" do
+      organisation = create(:delivery_partner_organisation)
+      get :index
+      expect(response).to redirect_to(root_path)
+      get :show, params: {id: organisation.id}
```

Comment:
I think you don't need to go to the "expense" of creating an organisation object and saving it to the db here and that something like

```
get :show, params: {id: `double`}
```

will do better as it:

- avoids the object creation/persistence
- makes it clear that the organisation ID is not relevant to the access control mechanism and redirection

Reply:
Nice! Thank you.